sharing_portal/views.py

```diff
@@ -273,39 +301,64 @@ def share_artifact(request, artifact):
             return HttpResponseRedirect(reverse('sharing_portal:detail', args=[artifact.pk]))
     else:
         form = ShareArtifactForm(
+            request,
             initial={
                 "is_public": artifact.is_public,
                 "projects": artifact.shared_to_projects.all(),
-            }
+                "is_reproducible": artifact.is_reproducible,
+                "project": artifact.project,
+                "reproduce_hours": artifact.reproduce_hours,
+            },
         )
         z_form = ZenodoPublishFormset(artifact_versions=artifact.versions)
 
     share_url = request.build_absolute_uri(
-        reverse('sharing_portal:detail', kwargs={'pk': artifact.pk}))
+        reverse("sharing_portal:detail", kwargs={"pk": artifact.pk})
+    )
     if artifact.sharing_key:
-        share_url += '?{key_name}={key_value}'.format(
-            key_name=SHARING_KEY_PARAM,
-            key_value=artifact.sharing_key)
+        share_url += "?{key_name}={key_value}".format(
+            key_name=SHARING_KEY_PARAM, key_value=artifact.sharing_key
+        )
 
-    template = loader.get_template('sharing_portal/share.html')
+    template = loader.get_template("sharing_portal/share.html")
     context = {
-        'share_form': form,
-        'z_management_form': z_form.management_form,
-        'z_forms': _artifact_display_versions(z_form.forms),
-        'share_url': share_url,
-        'artifact': artifact,
-        'artifact_json': JSONRenderer().render(ArtifactSerializer(instance=artifact).data),
+        "share_form": form,
+        "z_management_form": z_form.management_form,
+        "z_forms": _artifact_display_versions(z_form.forms),
+        "share_url": share_url,
+        "artifact": artifact,
+        "artifact_json": JSONRenderer().render(
+            ArtifactSerializer(instance=artifact).data
+        ),
     }
-
     return HttpResponse(template.render(context, request))
 
 
+def has_active_allocations(request):
+    mapper = ProjectAllocationMapper(request)
+    user_projects = mapper.get_user_projects(
+        request.user.username, to_pytas_model=False
+    )
+    for project in user_projects:
+        for allocation in project["allocations"]:
+            if allocation["status"].lower() == "active":
+                return True
+    return False
+
+
+def preserve_sharing_key(url, request):
+    if SHARING_KEY_PARAM in request.GET:
+        return url + "?{}={}".format(SHARING_KEY_PARAM, request.GET[SHARING_KEY_PARAM])
+    return url
+
+
 @check_view_permission
 def artifact(request, artifact, artifact_versions, version_idx=None):
```

Comment:
Cool, I wanted to make sure we still show the button for folks not logged in

Reply:
Yep!